src/flows/Onboarding/AnnualGrossSalary.tsx

```diff
@@ -0,0 +1,207 @@
+import { ReactNode, useState, useCallback, useRef, useEffect } from 'react';
+import { useFormContext } from 'react-hook-form';
+import debounce from 'lodash/debounce';
+import { TextField } from '@/src/components/form/fields/TextField';
+import { useConvertCurrency } from '@/src/flows/Onboarding/api';
+import { JSFField } from '@/src/types/remoteFlows';
+import { useFormFields } from '@/src/context';
+
+function useDebounce(
+  callback: (value: string) => Promise<void>,
+  delay: number,
+) {
+  const callbackRef = useRef(callback);
+
+  useEffect(() => {
+    callbackRef.current = callback;
+  }, [callback]);
+
+  const debouncedFn = useRef(
+    debounce((value: string) => {
+      callbackRef.current(value);
+    }, delay),
+  ).current;
+
+  return useCallback(
+    (value: string) => {
+      debouncedFn(value);
+    },
+    [debouncedFn],
+  );
+}
+
+type DescriptionWithConversionProps = {
+  description: ReactNode;
+  showConversion: boolean;
+  currency: string;
+  onClick: (evt: React.MouseEvent<HTMLButtonElement>) => void;
+};
+
+const DescriptionWithConversion = ({
+  description,
+  showConversion,
+  currency,
+  onClick,
+}: DescriptionWithConversionProps) => {
+  const { components } = useFormFields();
+  const label = showConversion
+    ? `Hide ${currency} conversion`
+    : `Show ${currency} conversion`;
+
+  const CustomButton = components?.button;
+
+  return (
+    <span className="RemoteFlows-AnnualGrossSalary-description">
+      {description}{' '}
+      {CustomButton ? (
+        <CustomButton
+          className="RemoteFlows-AnnualGrossSalary-button"
+          data-type="inline"
+          onClick={onClick}
+        >
+          {label}
+        </CustomButton>
+      ) : (
+        <button
+          className="RemoteFlows-AnnualGrossSalary-button"
+          onClick={onClick}
+        >
+          {label}
+        </button>
+      )}
+    </span>
+  );
+};
+
+type AnnualGrossSalaryProps = JSFField & {
+  currency: string;
+  desiredCurrency: string;
+  annual_gross_salary_conversion_properties?: {
+    label?: string;
+    description?: string;
+  };
+};
+
+export const AnnualGrossSalary = ({
+  currency,
+  desiredCurrency,
+  description,
+  ...props
+}: AnnualGrossSalaryProps) => {
+  const [showConversion, setShowConversion] = useState(false);
+  const { setValue, watch } = useFormContext();
+  const annualGrossSalary = watch(props.name);
+
+  const canShowConversion =
+    currency && desiredCurrency && currency !== desiredCurrency;
+
+  const { mutateAsync: convertCurrency } = useConvertCurrency();
+
+  const convertCurrencyCallback = useCallback(
```

Comment:
The code in this function is pretty much the same as `convertCurrencyReverseCallback` (only source_currency and target_currency change), so I think we should extract this code to a function and call it with the different parameters.